app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,23 @@
+class CommentsController < ApplicationController
+  def create
+    @post = Post.find(params[:post_id])
+    @comment = @post.comments.create!(text: comment_params[:text], author_id: current_user.id, post_id: @post.id)
+    respond_to do |format|
+      format.html do
+        if @comment.save
+          redirect_to user_post_comments_path(current_user, @comment.post_id)
+          flash[:success] = 'Comment successfully created'
+        else
+          render :create
+          flash.now[:error] = 'Error: Comment failed to save'
+        end
+      end
+    end
+  end
+
+  private
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
+end
```

Comment:
- [ ] Please make sure the comments controller should have new action similar to the one you created for the posts controller and also update the routes for comments with new.